impeller/geometry/gradient.cc

```diff
@@ -77,16 +76,17 @@ std::vector<uint8_t> CreateGradientBuffer(const std::vector<Color>& colors,
 
         AppendColor(mixed_color, &color_stop_channels);
       } else {
-        // We've slightly overshot the next stop. In theory this only happens if
-        // we have scaled our texture such that not every stop gets their own
-        // index. For now I am simply ignoring the inbetween colors. Currently
-        // this requires a gradient with either an absurd number of textures
-        // or very small stops.
-        AppendColor(next_color, &color_stop_channels);
-
+        // We've slightly overshot the previous stop.
```

Comment:
This isn't the case I though it was. If we've passed the previous stop, then we essentially just treat the current stop as the previous stop and run the same code as above.